internal/scanner/scanner.go

```diff
@@ -52,13 +54,57 @@ type Scanner struct {
 	printJSON  bool
 }
 
-// NewScanner creates a new scanner with the PoliciesFetcher provided
-func NewScanner(policiesFetcher PoliciesFetcher, resourcesFetcher ResourcesFetcher, printJSON bool) (*Scanner, error) {
+// NewScanner creates a new scanner with the PoliciesFetcher provided. If
+// insecureClient is false, it will read the caCertFile and add it to the in-app
+// cert trust store. This gets used by the httpCLient when connection to
+// PolicyServers endpoints.
+func NewScanner(policiesFetcher PoliciesFetcher, resourcesFetcher ResourcesFetcher,
+	printJSON bool,
+	insecureClient bool, caCertFile string) (*Scanner, error) {
 	report, err := report.NewPolicyReportStore()
 	if err != nil {
 		return nil, err
 	}
-	return &Scanner{policiesFetcher, resourcesFetcher, *report, http.Client{}, printJSON}, nil
+
+	// configure client for TLS
+	var httpClient http.Client
+	if insecureClient {
+		tr := &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: true}} //nolint
+		httpClient = http.Client{Transport: tr}
+		log.Info().Msg("connecting to PolicyServers endpoints in insecure SSL mode")
+	} else {
+		// Get the SystemCertPool, continue with an empty pool on error
+		rootCAs, _ := x509.SystemCertPool()
+		if rootCAs == nil {
+			rootCAs = x509.NewCertPool()
+		}
+		// Read in the cert file
+		certs, err := os.ReadFile(caCertFile)
+		if err != nil {
+			log.Fatal().Msg(fmt.Sprintf("failed to read file %q with CA cert: %v", caCertFile, err))
+		}
+		// Append our cert to the inherited system pool
+		if ok := rootCAs.AppendCertsFromPEM(certs); !ok {
+			log.Fatal().Msg("failed to append cert to in-app RootCAs trust store")
+		}
+		log.Debug().Str("ca-cert-file", caCertFile).
+			Msg("appended cert file to in-app RootCAs trust store")
+		// Trust the augmented cert pool in our client
+		config := &tls.Config{
+			InsecureSkipVerify: false,
+			RootCAs:            rootCAs,
+			MinVersion:         tls.VersionTLS12,
+		}
+		// initialize httpClient while conserving default settings
+		httpClient = *http.DefaultClient
+		httpClient.Transport = http.DefaultTransport
+		tr, ok := httpClient.Transport.(*http.Transport)
+		if !ok {
+			log.Fatal().Msg("failed to build httpClient: failed http.Transport type assertion")
+		}
+		tr.TLSClientConfig = config
+	}
+	return &Scanner{policiesFetcher, resourcesFetcher, *report, httpClient, printJSON}, nil
```

Comment:
Not related to this change, what is going to happen when the user provides a policy-server url that begins with `http` instead of `https`? Is the code going to work fine?

Reply:
Yes, it works. Tested it locally:
```
$ ./bin/audit-scanner -k kubewarden --policy-server-url http://localhost:3000 -l debug --insecure-ssl --extra-ca /home/vic/ca-cert 
{"level":"info","time":"2023-07-28T16:28:02+02:00","message":"querying PolicyServers at http://localhost:3000 for debugging purposes. Don't forget to start `kwctl port-forward` if needed"}
{"level":"debug","ca-cert-file":"/home/vic/ca-cert","time":"2023-07-28T16:28:02+02:00","message":"appended cert file to in-app RootCAs trust store"}
{"level":"warn","time":"2023-07-28T16:28:02+02:00","message":"connecting to PolicyServers endpoints without validating TLS connection"}
{"level":"info","time":"2023-07-28T16:28:02+02:00","message":"cluster wide scan started"}
...
```

Reply:
nice!